packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+        // The 0.2 here means 20% of the viewport width, which is set as
+        // the max width for the breadcrumb in the CSS style.
+        if (tooltipElement.current.offsetWidth / window.innerWidth >= 0.2)
+          setTooltipVisible(true);
+        else setTooltipVisible(false);
+      }
+    }
+    window.addEventListener('resize', updateTooltip);
+    updateTooltip();
+    return () => window.removeEventListener('resize', updateTooltip);
+  }, [tooltipElement, setTooltipVisible]);
+
+  return (
+    <Tooltip
+      ref={tooltipElement}
+      classes={classes}
+      PopperProps={{
+        popperOptions: {
+          modifiers: {
+            arrow: {
+              enabled: Boolean(arrowRef),
+              element: arrowRef,
+            },
+          },
+        },
+      }}
+      {...props}
+      title={
+        <React.Fragment>
+          {props.title}
+          <span className={arrow} ref={setArrowRef} />
+        </React.Fragment>
+      }
+      disableHoverListener={!isTooltipVisibile}
+    />
+  );
+};
+
+const styles = (): StyleRules =>
+  createStyles({
+    breadcrumb: {
+      display: 'block',
+      whiteSpace: 'nowrap',
+      maxWidth: '20vw',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+    },
+  });
+
+interface BreadcrumbProps {
+  apiUrl: string;
+  location: string;
+}
+
+interface Breadcrumb {
+  id: string;
+  pathName: string;
+  displayName: string;
+  url: string;
+}
+
+interface BreadcrumbState {
+  base: {
+    entityName: string;
+    displayName: string;
+    url: string;
+    isLast: boolean;
+  };
+  currentHierarchy: Breadcrumb[];
+  last: {
+    displayName: string;
+  };
+}
+
+interface WrappedProps extends WithStyles<typeof styles> {
+  displayName: string;
+  ariaLabel: string;
+  url?: string;
+  isLast?: boolean;
+}
+
+class WrappedBreadcrumb extends React.Component<
+  WrappedProps & WithStyles<typeof styles>
+> {
+  public constructor(props: WrappedProps & WithStyles<typeof styles>) {
+    super(props);
+  }
+
+  public render(): React.ReactElement {
+    return (
+      <ArrowTooltip title={this.props.displayName}>
+        {this.props.url ? (
+          <MaterialLink
+            component={Link}
+            to={this.props.url}
+            aria-label={this.props.ariaLabel}
+            className={this.props.classes.breadcrumb}
+          >
+            {this.props.displayName}
+          </MaterialLink>
+        ) : (
+          <Typography
+            color="textPrimary"
+            aria-label={this.props.ariaLabel}
+            className={this.props.classes.breadcrumb}
+          >
+            {this.props.isLast ? (
+              <i>{this.props.displayName}</i>
+            ) : (
+              this.props.displayName
+            )}
+          </Typography>
+        )}
+      </ArrowTooltip>
+    );
+  }
+}
+
+const StyledBreadcrumb = withStyles(styles)(WrappedBreadcrumb);
+
+class PageBreadcrumbs extends React.Component<
```

Comment:
This name should match the BreadcrumbProps/State name

Reply:
@louise-davies this has been updated now in https://github.com/ral-facilities/datagateway/pull/70/commits/93f854af36b7327f17c63fc1602ea46bf0ccb66b.

Reply:
@louise-davies resolved in https://github.com/ral-facilities/datagateway/pull/70/commits/93f854af36b7327f17c63fc1602ea46bf0ccb66b.